src/screens/Customers/CustomerMain.tsx

```diff
@@ -2,28 +2,31 @@ import React, { useEffect, useState } from 'react';
 import { CustomerRecord } from '../../utils/airtable/interface';
 import { getAllCustomers } from '../../utils/airtable/requests';
 import { Link, RouteComponentProps } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { RootState } from '../../lib/redux/store';
 
-function CustomerMain(props: RouteComponentProps) {
-  useEffect(() => {
-    getCustomers();
-  }, []);
-
-  const [customers, setCustomers] = useState<CustomerRecord[]>([]);
-
-  const getCustomers = async () => {
-    const customers: CustomerRecord[] = await getAllCustomers();
-    setCustomers(customers);
-  };
+interface CustomerProps {
+  match: { url: string; }
+  customers: CustomerRecord[];
+}
 
-  return (
-    <>
-      {customers.map((customer, index) => (
-        <Link key={index} to={{ pathname: `${props.match.url}/${customer.name}`, state: { customer: customer } }}>
-          <h3>{customer.name}</h3>
-        </Link>
-      ))}
-    </>
-  );
+class CustomerMain extends React.Component<CustomerProps, {}> {
```

Comment:
Is there a reason why you changed this from a functional component to a class component?